app/src/main/java/chat/rocket/android/authentication/presentation/AuthenticationNavigator.kt

```diff
@@ -128,12 +139,25 @@ class AuthenticationNavigator(internal val activity: AuthenticationActivity) {
     }
 
     fun toChatList() {
-        activity.startActivity(Intent(activity, MainActivity::class.java))
-        activity.finish()
+        savedDeepLinkInfo?.let {
+            toChatList(it)
+        } .ifNull {
+            activity.startActivity(Intent(activity, MainActivity::class.java))
+            activity.finish()
+        }
+        savedDeepLinkInfo = null
     }
 
     fun toChatList(serverUrl: String) {
         activity.startActivity(activity.changeServerIntent(serverUrl))
         activity.finish()
     }
+
+    fun toChatList(deepLinkInfo: DeepLinkInfo) {
```

Comment:
I think that comments here would help to explain the use cases for these three different constructors, otherwise it's kind of opaque.

Reply:
These are not constructors? But if you want I can consolidate toChatList to just a single method.